src/common/connection.js

```diff
@@ -441,6 +441,7 @@ class Connection extends EventEmitter {
       this._whenReady(this._send(message)).then(() => {
         const delay = timeout || this._timeout
         timer = setTimeout(() => _reject(new TimeoutError()), delay)
+        timer.unref()
```

Comment:
As long as we wrap this in a conditional, this should be fine on both Node.js and the browser:

```
       // In Node, tell Node not to leave the process open if this timeout is  keep the event loop active waiting only for this timer.
        if (timer.unref) {
          timer.unref();
        }
```

Also, can you confirm that the process doesn't exit before the response has returned (since this timeris no longer keeping it open)